googlespreadsheet/src/org/wso2/ballerina/connectors/googlespreadsheet/googlespreadsheetConnector.bal

```diff
@@ -0,0 +1,865 @@
+
+    @doc:Description ("Retrieve Sheet properties and other metadata")
+    @doc:Param("spreadsheet: The Google Spreadsheet Connector instance")
+    @doc:Param("spreadsheetId: Unique value of the spreadsheet")
+    @doc:Param("includeGridData: True if grid data should be returned. This parameter is ignored
+                                    if a field mask was set in the request")
+    @doc:Param("fields: Specifying which fields to include in a partial response")
+    @doc:Param("ranges: The ranges to retrieve from the spreadsheet")
+    @doc:Return ("response object")
+    action getSheetMetaData (ClientConnector spreadsheet, string spreadsheetId, string includeGridData, string fields,
+                             string ranges) (message) {
+
+        message request = {};
+        string uriParams;
+        string getSheetMetaDataPath = "/v4/spreadsheets/" + spreadsheetId;
+
+        if (includeGridData != "null") {
+            uriParams = uriParams + "&includeGridData=" + includeGridData;
+        }
+
+        if (fields != "null") {
+            uriParams = uriParams + "&fields=" + fields;
+        }
+
+        while(strings:indexOf(ranges, ",") != -1) {
+            int index = strings:indexOf(ranges, ",");
+            uriParams = uriParams + "&ranges=" + strings:subString(ranges, 0, index);
+            ranges = strings:subString(ranges, index+1, strings:length(ranges));
+        }
+
+        if(ranges != "null") {
+            uriParams = uriParams + "&ranges=" + ranges;
+        }
+
+        if(uriParams != "") {
+            getSheetMetaDataPath = getSheetMetaDataPath + "?" + strings:subString(
+                                                                        uriParams, 1, strings:length(uriParams));
+        }
+
+        message response = oauth2:ClientConnector.get(googlespreadsheetEP, getSheetMetaDataPath, request);
+        return response;
+    }
+
+    @doc:Description("Copies a single sheet from a spreadsheet to another spreadsheet")
+    @doc:Param("spreadsheet: The Google Spreadsheet Connector instance")
+    @doc:Param("spreadsheetId: Unique value of the spreadsheet")
+    @doc:Param("sheetId: The ID of the sheet to copy")
+    @doc:Param("payload: It contains the ID of the spreadsheet to copy the sheet to")
+    @doc:Param("fields: Specifying which fields to include in a partial response")
+    @doc:Return("response object")
+    action copyTo(ClientConnector spreadsheet, string spreadsheetId, string sheetId,
+                  json payload, string fields) (message) {
+
+        message request = {};
+        string uriParams;
+        string copyToPath = "/v4/spreadsheets/" + spreadsheetId + "/sheets/" + sheetId + ":copyTo";
+
+        if(fields != "null") {
+            uriParams = uriParams + "fields=" + fields;
+        }
+
+        if(uriParams != "") {
+            copyToPath = copyToPath + "?" + uriParams;
+        }
+
+        messages:setHeader(request, "Content-Type", "Application/json");
```

Comment:
messages:setJsonPayload function sets the content-type header to Application/json. Therefore there is no need to add the header manually.